src/main/kotlin/no/skatteetaten/aurora/boober/feature/S3Feature.kt

```diff
@@ -276,6 +281,17 @@ private fun List<S3BucketObjectArea>.verifyObjectAreaAndBucketAreUnique(): List<
         }
 }
 
+private fun List<S3BucketObjectArea>.validateTenantContainsCurrentAffiliationPrefix(affiliation: String): List<IllegalArgumentException> {
```

Comment:
Kunne skrevet denne slik for eksempel:
```
private fun List<S3BucketObjectArea>.validateTenantContainsCurrentAffiliationPrefix(affiliation: String): List<IllegalArgumentException> {
    return this.filterNot {
        it.tenant.startsWith(affiliation)
    }.map {
        IllegalArgumentException("tenant must contain current affiliation=$affiliation as a prefix, specified value was: ${it.tenant}")
    }
}
```

Det er ikke nødvendig med flatmap her
Et annet alternativ er å skrive den slik:
```
private fun List<S3BucketObjectArea>.validateTenantContainsCurrentAffiliationPrefix(affiliation: String): List<IllegalArgumentException> {
    return this.mapNotNull {
        if (!it.tenant.startsWith(affiliation)) {
            IllegalArgumentException("tenant must contain current affiliation=$affiliation as a prefix, specified value was: ${it.tenant}")
        } else null
    }
}
```

Du ser hvilken du foretrekker

Reply:
Enig, trenger ikke flatmap her nei. Endret til det første forslaget ditt